lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -9036,11 +9032,23 @@ namespace Js
             length = pArr->length;
         }
 
+        if (length.IsSmallIndex())
```

Comment:
Yes, that's true. If the length is a large index, though, we're going to walk across the entire 32-bit range and at least call HasItem for each one which would take hours. We could split the walk into two pieces - one for small index values and one for any big index values - but this is an edge case and I don't think it's warranted.